pkg/v1/tkg/web/src/app/views/landing/wizard/shared/wizard-base/wizard-base.ts

```diff
@@ -343,14 +347,23 @@ export abstract class WizardBaseDirective extends BasicSubscriber implements Wiz
 
     /**
      * @param arrObj of type [ {key: 'a', value: '1}, {key : 'b, value:'2}]
-     * @param key To Identify the ObjectKey
-     * @param value To Identify the ObjectValue
+     * @param params.key<Function> To Identify the ObjectKey
+     * @param params.value<Function> To Identify the ObjectValue
      * @return Object {'a': 1 , 'b': '1'}
      */
-    arrayOfObjectsToObject<T>(arrObj: T[], params: Params<T>): { [key: string]: string; } {
-        return arrObj && arrObj instanceof Array
-            ? arrObj.reduce((obj, item) => ((obj[params.key(item)] = params.value(item)), obj), {})
-            : {};
+    arrayOfObjectsToObject<T>(arrObj: T[], params: Params<T>): KeyValueObject {
+        if (!arrObj || !(arrObj instanceof Array)) {
+            return {};
+        }
+
+        return arrObj.reduce<KeyValueObject>(
```

Comment:
nit: Either here and/or in the comment block above, might be good to add a comment that we DROP any object with a blank key

Reply:
Good idea, will do. Thank you!